### ISSUE TITLE
Explain deletions in brew upgrade

### ISSUE THREAD

**Issue by ralphrmartin** [Homebrew/brew#11341]
### Provide a detailed description of the proposed feature

When a formula or cask is deleted, it would be helpful to provide some brief explanation to users as to why that has been done when they see that on doing brew upgrade. I'm not suggesting anything lengthy, just a single sentence along the lines of 
Cask gobbledegook has been deleted because of licencing issues
Formula thingummybob has been deleted as it has been superseded by formula newwhatsit
Cask doodah has been deleted because no one has maintained it in a long time and it no longer works 

### What is the motivation for the feature?

It will help users decide how to proceed in overcoming their difficulties in the software no longer being available, or to know whether a suitable replacement exists in homebrew.

### How will the feature be relevant to at least 90% of Homebrew users?

All users may see such xxx has been deleted messages.

### What alternatives to the feature have been considered?

The current alternative, just seeing 
xxx has been deleted
is unhelpful: it tells users something they were relying on has been taken away, without giving them any clue why, or what they might be able to do about it.

**Comment by iMichka** [comment]
For homebrew-core, we nowadays disable the formula for 12 months and then only it gets deleted. We also have a deprecation warning in place for some others. This leaves you time to be aware of the upcoming change.
Of course sometimes we might delete something in emergency, but that's quite rare.
If you want the deletion reason because you notice this after the 12 month period, you'll have to look at the git commits or the related pull requests.

For casks I don't know how they handle this right now, but it should not be that hard to follow the same system on their side.

The issue is that all the information we have of a formula is in the code that is checked out on your computer. Once the package is gone, the information is deleted from you computer, so there is nowhere to store the deletion reasons right now.

**Comment by ralphrmartin** [comment]
"If you want the deletion reason because you notice this after the 12 month period, you'll have to look at the git commits or the related pull requests."

Please try to look at this from the point of view of ordinary users, not a developer - surely the whole point of homebrew was to make installing software easier for ordinary users? Disabling a formula is not made obvious to a user who already has it installed. In most cases, they don't see any warning when doing brew upgrade (although I did see some concerning macfuse stuff), just a very stark "formula deleted" message at the time of deletion, and that has no explanation as to why this has happened. Many ordinary users will not be familiar with github or how it works.

As a minimal solution, you could add a link to the git commits or pull requests in the deletion announcement. Ordinary users are not going to be aware this is a place to look.

**Comment by MikeMcQuaid** [comment]
> Please try to look at this from the point of view of ordinary users, not a developer - surely the whole point of homebrew was to make installing software easier for ordinary users?

Homebrew users are developers. Homebrew was designed to be used by developers. That's why our support tracker is on GitHub and not e.g. Zendesk.

---

Passing on this. We'll consider PRs to improve automation around this but as-is this is "maintainers: please do more work".
